packages/@webex/plugin-cc/src/cc.ts

```diff
@@ -1,33 +1,139 @@
+} from './types';
+import {EVENT, READY, WEBSOCKET_EVENT_TIMEOUT, SUBSCRIBE_API, WCC_API_GATEWAY} from './constants';
+import IWebSocket from './WebSocket/types';
+import WebSocket from './WebSocket/WebSocket';
+
+const REGISTER_EVENT = 'register';
 
 export default class ContactCenter extends WebexPlugin implements IContactCenter {
-  namespace = 'ContactCenter';
+  namespace = 'cc';
   $config: CCPluginConfig;
   $webex: WebexSDK;
   wccApiUrl: string;
+  webSocket: IWebSocket;
+  ciUserId: string;
+  registered = false;
+  eventHandlers: Map<
+    string,
+    {resolve: (data: any) => void; reject: (error: any) => void; timeoutId: NodeJS.Timeout}
+  > = new Map();
 
   constructor(...args) {
     super(...args);
-    // @ts-ignore
-    this.$config = this.config;
+
     // @ts-ignore
     this.$webex = this.webex;
+
+    this.$webex.once(READY, () => {
+      // @ts-ignore
+      this.$config = this.config;
+      this.webSocket = new WebSocket({
+        parent: this.$webex,
+      });
+    });
   }
 
-  register(success: boolean): Promise<string> {
-    this.wccApiUrl = this.$webex.internal.services.get(WCC_API_GATEWAY); // Added this change for Ravi's PR, he will move this under different function if needed.
-    // TODO: Mercury Subsciption code should be added as part of this function
+  /**
+   * This is used for making the CC SDK ready by setting up the cc mercury connection.
+   */
+  public async register(): Promise<string> {
+    this.wccApiUrl = this.$webex.internal.services.get(WCC_API_GATEWAY);
+
+    this.listenForWebSocketEvents();
 
     return new Promise((resolve, reject) => {
-      setTimeout(() => {
-        if (success) {
-          resolve('Success: Dummy data returned');
-        } else {
-          reject(new Error('Simulated Error'));
-        }
-      }, 1000);
+      this.addEventHandler(
+        REGISTER_EVENT,
+        (result) => {
+          this.registered = true;
+          resolve(result);
+        },
+        reject
+      );
+
+      this.establishConnection(reject);
+    });
+  }
+
+  /**
+   * This is used for unregistering the CC SDK by disconnecting the cc mercury connection.
+   * @returns Promise<void>
+   */
+  public unregister(): Promise<void> {
+    return this.webSocket.disconnectWebSocket().then(() => {
+      this.webSocket.off(EVENT, this.processEvent);
+      this.registered = false;
+    });
+  }
+
+  private listenForWebSocketEvents() {
+    this.webSocket.on(EVENT, this.processEvent);
+  }
+
+  private processEvent = (event: WebSocketEvent): void => {
+    switch (event.type) {
+      case CC_EVENTS.WELCOME:
+        this.ciUserId = event.data.agentId;
+        this.handleEvent(REGISTER_EVENT, `Success: CI User ID is ${this.ciUserId}`); // TODO: Will send AgentPRofile object as part of Parv's PR
+        break;
+      default:
+        this.$webex.logger.info(`Unknown event: ${event.type}`);
+    }
+  };
+
+  private async establishConnection(reject: (error: any) => void) {
```

Comment:
Can we avoid using any here as well. We can use the alternative here to use .then() and .catch so that we don't have to handle reject and pass it in as argument

Reply:
addressed